packages/preset-mini/src/_rules/variables.ts

```diff
@@ -26,14 +26,21 @@ export const cssVariables: Rule[] = [
 ]
 
 export const cssProperty: Rule[] = [
-  [/^\[(.*)\]$/, ([_, body]) => {
+  [/^\[(.*)\]$/, ([_, body], { theme }) => {
     if (!body.includes(':'))
       return
 
     const [prop, ...rest] = body.split(':')
     const value = rest.join(':')
+
     if (!isURI(body) && prop.match(/^[a-z-]+$/) && isValidCSSBody(value)) {
-      const parsed = h.bracket(`[${value}]`)
+      let parsed
+
+      if (hasThemeFn(value))
+        parsed = transformThemeFn(value, theme)
+      else
+        parsed = h.bracket(`[${value}]`)
```

Comment:
```suggestion
      if (!parsed)
        parsed = h.bracket(`[${value}]`)
```

So here we could fallback if the theme fn ends up invalid by the regex.

Reply:
Can we use https://github.com/unocss/unocss/pull/3204/commits/5d53238ae5ef652ad7046c7279b0bc4e6d933eee instead?